game-frontend/src/service/game/context.ts

```diff
@@ -34,11 +37,30 @@ export function useWebsocket(): ContextState {
   useEffect(() => {
     if (socket) {
       socket.addEventListener('message', event => {
-        const action: Message = JSON.parse(event.data);
-        dispatch(action);
+        const message: Message = JSON.parse(event.data);
+        dispatch(message);
       });
+      // persist role choice
+      socket.addEventListener('message', event => {
```

Comment:
No need to add a different event listener. This should be handled within the other event listener.